src/App.tsx

```diff
@@ -1,14 +1,56 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { FilterStatus } from './types/FilterStatus';
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoFilter } from './components/TodoFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [todoId, setTodoId] = useState(0);
+  const [filter, setFilter] = useState<FilterStatus>(FilterStatus.ALL);
+  const [responseFilter, setResponseFilter] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then((response) => {
+        setTodos(response);
+        setLoading(false);
+      });
+  }, []);
+
+  const stringIncludes = (full: string, part: string) => (
+    full.toLowerCase().includes(part.toLowerCase())
+  );
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case FilterStatus.COMPLETED:
+        return (
+          todos
+            .filter(todo => todo.completed)
+            .filter(todo => stringIncludes(todo.title, responseFilter))
```

Comment:
any way you filter two times )
```suggestion
          todos
            .filter(todo => todo.completed && stringIncludes(todo.title, responseFilter))
```